triton-client/src/components/DatasetList/index.tsx

```diff
@@ -1,65 +1,105 @@
-import { Divider, Empty, Spin } from "antd"
-import { useMemo } from "react"
-import { useAppSelector } from "@store/hooks"
-import DatasetCard from "@components/DatasetCard"
+import { Table, TableProps } from "antd"
+import { useCallback, useEffect, useState } from "react"
+import { useAppDispatch } from "@store/hooks"
 import { DatasetListProps } from "./interfaces"
+import { fetchDatasets, fetchReadsets, fetchRequests } from "@store/thunks"
+import {
+    DatasetColumnSource,
+    useDatasetColumns,
+} from "@components/DatasetColumns"
+import { TritonDataset, TritonRequest } from "@api/api-types"
 
-export default function DatasetList({ runName }: DatasetListProps) {
-    const datasetIDs = useAppSelector(
-        (state) => state.runsState.runsByName[runName]?.datasets,
-    )
-    const datasetsByID = useAppSelector(
-        (state) => state.datasetsState.datasetsById,
-    )
+export default function DatasetList({ externalProjectID }: DatasetListProps) {
+    const dispatch = useAppDispatch()
+    const [isFetching, setIsFetching] = useState(true)
+    const [dataSource, setDataSource] = useState<
+        NonNullable<TableProps<DatasetColumnSource>["dataSource"]>
+    >([])
 
-    const renderDatasets = useMemo(() => {
-        if (datasetIDs === undefined) return [<Spin key={"spin"} />]
-        if (datasetIDs.length === 0)
-            return [
-                <Empty
-                    key={"empty"}
-                    description={
-                        "There are no sample data available for request."
-                    }
-                />,
-            ]
-        if (datasetIDs.length > 0) {
-            console.info(datasetIDs, datasetsByID)
+    useEffect(() => {
```

Comment:
this single useEffect hook function has a lot of logic in it. it would be good to create functions in order to compartmentalize the logic into their own functions (the same approach as with the components but at the function level)